CAN_Decoder_SourceCode_TestRel_21Dec21.py

```diff
@@ -0,0 +1,119 @@
+import tkinter as tk
+from tkinter import *
+from tkinter import filedialog
+from datetime import datetime
+import re
+
+
+def CAN_Decoder():
+    dataFile = filedialog.askopenfilename()
+    inputFrameData = []
+    
+    # Extract frame from text file using regex approach
+    typicalFrameForm = r"can(\d)\s+([(\w)]{1,3})\s+\[(\d+)\]\s+([\w]{2}(?:\s[\w]{2})*)"
+    frameSize = 0
+    for dataFrame in re.findall(typicalFrameForm, open(dataFile).read()):
+        diagnosticID = int(dataFrame[1], 16)
+        data = list(map(lambda x: int(x, 16), dataFrame[3].split(" ")))
+        frameType = data[0]
+        if frameType <= 6:
+            frameSize = frameType  # Single frame
+        elif frameType == 16:  # 10 for hex notation
```

Comment:
based on ISO-TP documentation if the frame starts with `1A BC`  the total frame size will be `ABC`. In these examples `A` just happended to be always 0. I will let you figure out a way to extract `A` from first byte and append it to `BC`